cli/js/testing.ts

```diff
@@ -83,14 +82,134 @@ export interface RunTestsOptions {
   only?: string | RegExp;
   skip?: string | RegExp;
   disableLog?: boolean;
+  reporter?: TestReporter;
+}
+
+type FilterFn = (testDef: TestDefinition) => boolean;
+
+interface TestResult {
+  passed: boolean;
+  name: string;
+  fn: TestFunction;
+  skipped: boolean;
+  hasRun: boolean;
+  duration: number;
+  error?: Error;
+}
+
+enum MsgKind {
+  Start = "start",
+  Test = "test",
+  End = "end"
+}
+
+interface StartMsg {
+  kind: MsgKind.Start;
+  tests: number;
+  stats: TestStats;
+}
+
+interface TestMsg {
+  kind: MsgKind.Test;
+  result: TestResult;
+}
+
+interface EndMsg {
+  kind: MsgKind.End;
+  stats: TestStats;
+  duration: number;
+  results: TestResult[];
+}
+
```

Comment:
Aside: it would be very nice if we could import the interface definitions from the d.ts file rather than duplicating them here.

Reply:
Not sure if that's possible, CC @kitsonk

Reply:
Tried importing `lib.deno.ns.d.ts` inside `cli/js/testing.ts` - it spawns plethora of TS errors